.github/workflows/firmware-unit-tests.yml

```
@@ -146,3 +203,63 @@ jobs:
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          cmake -B build/tidy \
+            -DCMAKE_BUILD_TYPE=Debug \
+            -DENABLE_CLANG_TIDY=ON \
+            -DCMAKE_C_COMPILER=clang-18
+
+      - name: Build (violations surface as compiler errors)
+        working-directory: star-rx72n-firmware/tests
+        # pipefail: propagate cmake exit code through the tee pipe so violations fail the step.
+        run: |
```

actionlint (1.7.11)

[error] shellcheck reported issue in this script: SC2046:warning:2:28: Quote this to prevent word splitting

(shellcheck)